### PR TITLE
Related tags v3

### PR DESCRIPTION
**Pull request by i-like-robots**
- Switches to V3 ES index
- Can now fetch related metadata ID and taxonomy from nested document

So given:

```js
related({
  taxonomies: [ 'topics', 'sections', 'people', 'organisations' ],
  prefLabel: 'United States of America',
  range: 60,
  size: 12
})
```

The following query will be constructed:

```json
{
  "size": 0,
  "query": {
    "bool": {
      "must": [
        {
          "term": {
            "metadata.prefLabel": "United States of America"
          }
        },
        {
          "range": {
            "publishedDate": {
              "gte": "2015-12-10T12:05:07.787Z",
              "lt": "2016-02-07T11:25:24.846Z"
            }
          }
        }
      ]
    }
  },
  "aggs": {
    "tags": {
      "nested": {
        "path": "metadata"
      },
      "aggs": {
        "filtered": {
            "filter": {
                "bool": {
                    "should": [
                        { "term": { "taxonomy": "topics" } },
                        { "term": { "taxonomy": "sections" } },
                        { "term": { "taxonomy": "people" } },
                        { "term": { "taxonomy": "organisations" } }
                    ]
                }
            },
            "aggs": {
                "suggestions": {
                    "terms": {
                        "field": "prefLabel",
                        "exclude": "United States of America",
                        "size": 12,
                        "min_doc_count": 1
                    },
                    "aggs": {
                        "id": {
                          "terms": {
                            "field": "idV1",
                            "size": 1
                          }
                        },
                        "taxonomy": {
                          "terms": {
                            "field": "taxonomy",
                            "size": 1
                          }
                        }
                    }
                }
            }
        }
      }
    }
  }
}
```

Which returns something like:

```js
[ { prefLabel: 'Markets',
    idV1: 'NzE=-U2VjdGlvbnM=',
    taxonomy: 'sections' },
  { prefLabel: 'Companies',
    idV1: 'Mjk=-U2VjdGlvbnM=',
    taxonomy: 'sections' },
  { prefLabel: 'Central Banks',
    idV1: 'NQ==-VG9waWNz',
    taxonomy: 'topics' },
  { prefLabel: 'World',
    idV1: 'MQ==-U2VjdGlvbnM=',
    taxonomy: 'sections' },
  { prefLabel: 'US & Canadian Companies',
    idV1: 'NjU=-U2VjdGlvbnM=',
    taxonomy: 'sections' },
  { prefLabel: 'US downturn',
    idV1: 'NDQ=-VG9waWNz',
    taxonomy: 'topics' },
  { prefLabel: 'Federal Reserve',
    idV1: 'TnN0ZWluX09OX0FGVE1fT05fMTIxNjIw-T04=',
    taxonomy: 'organisations' },
  { prefLabel: 'Mergers & Acquisitions',
    idV1: 'Mjk=-VG9waWNz',
    taxonomy: 'topics' },
  { prefLabel: 'Equities',
    idV1: 'OTg=-U2VjdGlvbnM=',
    taxonomy: 'sections' },
  { prefLabel: 'Oil',
    idV1: 'ZmFmYTUxOTItMGZjZC00YmJkLWJlZTQtMmY3ZDZiOWZkYmYw-VG9waWNz',
    taxonomy: 'topics' },
  { prefLabel: 'US & Canada',
    idV1: 'Mg==-U2VjdGlvbnM=',
    taxonomy: 'sections' },
  { prefLabel: 'Financials',
    idV1: 'NTc=-U2VjdGlvbnM=',
    taxonomy: 'sections' } ]
```

### CODE REVIEW

File: src/related-tags.js [hunk 1]
```diff
@@ -25,99 +18,101 @@ function formatDateRange(from, to) {
...
+					relatedTags: {
+						terms: {
+							field: 'prefLabel',
+							exclude: options.name,
+							min_doc_count: 2
+						},
+						aggs: {
+							id: {
+								terms: {
+									field: 'idV1',
```
Comment (i-like-robots) [inline src/related-tags.js]: These aggregations are run in the context of the parent so these fields are always the right ones =)

File: src/related-tags.js [hunk 2]
```diff
@@ -25,99 +18,101 @@ function formatDateRange(from, to) {
...
 		},
-		sort: {
-			'item.lifecycle.initialPublishDateTime': {
-				order: 'desc'
+		aggs: {
+			documentsWithTag: {
+				nested: {
+					path: 'item.metadata'
+				},
+				aggs: {
```
Comment (i-like-robots) [inline src/related-tags.js]: A filter could be added to limit tags by certain taxonomies

File: src/related-tags.js [hunk 3]
```diff
@@ -25,99 +18,101 @@ function formatDateRange(from, to) {
...
-					min_doc_count: 2
-				}
-			}
-		});
-	}, {});
-}
-
 function formatPostBody(options) {
 	return {
+		// Signed fetch does not allow query strings so no ?search_type=count
```
Comment (matthew-andrews) [inline src/related-tags.js]: oh really???

File: src/related-tags.js [hunk 4]
```diff
@@ -25,99 +18,101 @@ function formatDateRange(from, to) {
...
+				order: 'desc'
+			}
+		},
 		query: {
 			bool: {
 				must: [
 					{
-						match: formatSearchTerm(options)
+						term: {
+							'item.metadata.prefLabel': options.name
```
Comment (matthew-andrews) [inline src/related-tags.js]: item?

**Comment by matthew-andrews** [comment]
:+1:

**Pull request merged**